### ISSUE TITLE
Fix FN S2259 (`null-dereference`): when property is called `length`

### ISSUE THREAD

**Issue by saberduck** [SonarSource/SonarJS#2670]
We don't detect the issue for S2259 (null-dereference) when property is called `length`

```
 var x;
  if (x === undefined) {
      console.log(x.length); // FN
  }
```

**Comment by neoligero** [comment]
Hi @saberduck , any update on this?
I'd like to add that this issue not only happens with .length, Sonarlint doesn't detect not declared variables either.
Example:
var x;
  if (x === undefined) {
      console.log(x); // FN
      console.log(asdf); // Should be detected too
  }
This is a big issue for me since we are trying to use this as a standard in the team, but seems to be not possible atm.
Thanks.

**Comment by ilia-kebets-sonarsource** [comment]
For some reason, [isUndefinedOrNull()](https://github.com/SonarSource/SonarJS/blob/4e803a3f4c97e8c3ee8d519caf9d43350e4c172e/packages/jsts/src/rules/helpers/type.ts#L138-L144) returns `false` when the property is `.length`, and `true` when it's anything else. So it breaks the function of the [CheckNullDereference()](https://github.com/SonarSource/SonarJS/blob/4e803a3f4c97e8c3ee8d519caf9d43350e4c172e/packages/jsts/src/rules/S2259/rule.ts#L143-L173) function

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-130
